src/jdk.httpserver/share/classes/sun/net/httpserver/simpleserver/FileServerHandler.java

```diff
@@ -250,6 +254,36 @@ private static Path indexFile(Path path) {
         return Files.exists(html) ? html : Files.exists(htm) ? htm : null;
     }
 
+    private static boolean isFavIconRequest(HttpExchange exchange) {
+        return "/favicon.ico".equals(exchange.getRequestURI().getPath());
+    }
+
+    private void serveDefaultFavIcon(HttpExchange exchange, boolean writeBody)
+            throws IOException
+    {
+        var respHdrs = exchange.getResponseHeaders();
+        var icon = "/sun/net/httpserver/simpleserver/resources/favicon.ico";
+        try (var stream = getClass().getModule().getResourceAsStream(icon)) {
+            var bytes = stream.readAllBytes();
+            respHdrs.set("Content-Type", "image/x-icon");
+            // TODO respHdrs.set("Last-Modified", getLastModified(...));
```

Comment:
Is there a canonical way to retrieve a stable timestamp for the built-in `favicon.ico` here?

Reply:
Using Java's birthday `"Mon, 23 May 1995 11:11:11 GMT"` for the time being.

Reply:
[java.version.date](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/System.html#java.version.date) could be a reasonable alternative

Reply:
I thought about that too, but dates from the future might lead to other issues like caching.

`java -XshowSettings:properties --version`
```
...
java.version.date = 2024-09-17
...
openjdk 23-ea 2024-09-17
OpenJDK Runtime Environment (build 23-ea+21-1723)
OpenJDK 64-Bit Server VM (build 23-ea+21-1723, mixed mode, sharing)
```